core/transaction.go

```diff
@@ -340,53 +340,25 @@ func deployAccountTransactionHash(d *DeployAccountTransaction, n utils.Network)
 	return nil, errInvalidTransactionVersion(d, d.Version)
 }
 
-type CantVerifyTransactionHashError struct {
-	t           Transaction
-	hashFailure error
-	next        *CantVerifyTransactionHashError
-}
-
-func (e CantVerifyTransactionHashError) Unwrap() error {
-	if e.next != nil {
-		return *e.next
+func VerifyTransactions(txs []Transaction, n utils.Network, protocolVersion string) error {
+	blockVersion, err := ParseBlockVersion(protocolVersion)
+	if err != nil {
+		return err
 	}
-	return nil
-}
-
-func (e CantVerifyTransactionHashError) Error() string {
-	return fmt.Sprintf("cannot verify transaction hash of Transaction Type: %v: %v", reflect.TypeOf(e.t), e.hashFailure.Error())
-}
-
-func (e CantVerifyTransactionHashError) Hash() *felt.Felt {
-	return e.t.Hash()
-}
 
-func VerifyTransactions(txs []Transaction, n utils.Network) error {
-	var head *CantVerifyTransactionHashError
-	for _, tx := range txs {
-		if err := verifyTransactionHash(tx, n); err != nil {
-			err.next = head
-			head = err
-		}
-	}
-	if head != nil {
-		return *head
+	// blockVersion < 0.11.0
+	// only start verifying transaction hashes after 0.11.0
+	if blockVersion.Compare(semver.MustParse("0.11.0")) == -1 {
```

Comment:
Please store the version as a `const`.

Reply:
Sadly, `semver.MustParse` can not be assigned to a const